tests/utils/megaFormSplitAtom.test.tsx

```diff
@@ -0,0 +1,147 @@
+import { atom, Provider, useAtom, PrimitiveAtom } from 'jotai'
+import { focusAtom } from '../../src/optics'
+import React, { useMemo, useEffect, useRef } from 'react'
+import { render, fireEvent, waitFor } from '@testing-library/react'
+import { useAtomCallback } from '../../src/utils'
+import { splitAtom } from '../../src/utils/splitAtom'
+
+type TodoItem = { task: string; checked?: boolean }
+
+it('no unneccesary updates when updating atoms', async () => {
+  const formListAtom = atom<Record<string, Record<string, string>>>({
+    form: { hello: 'world' },
+  })
+  const utils = render(
+    <Provider>
+      <FormList formListAtom={formListAtom} />
+    </Provider>
+  )
+
+  const whatsup = await utils.findByTestId('world')
+  fireEvent.change(whatsup, { target: { value: 'some change' } })
+})
+
+const Field = ({
+  fieldAtom,
+  removeField,
+}: {
+  fieldAtom: PrimitiveAtom<{ name: string; value: string }>
+  removeField: () => void
+}) => {
+  const [name, setName] = useAtom(focusAtom(fieldAtom, (o) => o.prop('name')))
```

Comment:
```suggestion
  const [name, setName] = useAtom(useMemo(() => focusAtom(fieldAtom, (o) => o.prop('name')), [fieldAtom]))
```

Does this fix the test?

Reply:
Yes this fixes the test - (if applied to both useAtoms in that component) however I was hoping that we would not need to do `useMemo` since the value that the atom points to is referentially stable (`string`).

Reply:
jotai's concept totally depends on atom referential identity. the atom value being string doesn't help, unfortunately.